Source/WebCore/style/Styleable.cpp

```diff
@@ -735,8 +735,34 @@ void Styleable::updateCSSTransitions(const RenderStyle& currentStyle, const Rend
     if (currentStyle.display() == DisplayType::None)
         return;
 
-    // In case this element is newly getting a "display: none" we need to cancel all of its transitions and disregard new ones.
-    if (currentStyle.hasTransitions() && currentStyle.display() != DisplayType::None && newStyle.display() == DisplayType::None) {
+    auto transitionsDisplay = [](const RenderStyle& style) {
+        if (!style.hasTransitions())
+            return false;
+
+        for (auto& transition : *style.transitions()) {
+            auto transitionProperty = transition->property();
+            switch (transitionProperty.mode) {
+            case Animation::TransitionMode::All:
+                if (transition->allowsDiscreteTransitions())
+                    return true;
+                break;
+            case Animation::TransitionMode::SingleProperty:
+                if (std::holds_alternative<CSSPropertyID>(transitionProperty.animatableProperty)) {
+                    if (std::get<CSSPropertyID>(transitionProperty.animatableProperty) == CSSPropertyDisplay)
+                        return transition->allowsDiscreteTransitions();
```

Comment:
Does this work as expected if you have `transition: display 2s, all allow-discrete 3s;` ? Would it return false because it can't find allow-discrete in the first transition?

Reply:
Correct, good catch. I'll fix the patch and add an additional test for that case.